ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,353 @@
+
+    cuts_file = traits.Path(
+        default_value=None, directory_ok=False, help="Path to optimised cuts input file"
+    ).tag(config=True)
+
+    gamma_file = traits.Path(
+        default_value=None, directory_ok=False, help="Gamma input filename and path"
+    ).tag(config=True)
+    gamma_sim_spectrum = traits.UseEnum(
+        Spectra,
+        default_value=Spectra.CRAB_HEGRA,
+        help="Name of the pyrif spectra used for the simulated gamma spectrum",
+    ).tag(config=True)
+    proton_file = traits.Path(
+        default_value=None, directory_ok=False, help="Proton input filename and path"
+    ).tag(config=True)
+    proton_sim_spectrum = traits.UseEnum(
+        Spectra,
+        default_value=Spectra.IRFDOC_PROTON_SPECTRUM,
+        help="Name of the pyrif spectra used for the simulated proton spectrum",
+    ).tag(config=True)
+    electron_file = traits.Path(
+        default_value=None, directory_ok=False, help="Electron input filename and path"
+    ).tag(config=True)
+    electron_sim_spectrum = traits.UseEnum(
+        Spectra,
+        default_value=Spectra.IRFDOC_ELECTRON_SPECTRUM,
+        help="Name of the pyrif spectra used for the simulated electron spectrum",
+    ).tag(config=True)
+
+    chunk_size = Integer(
+        default_value=100000,
+        allow_none=True,
+        help="How many subarray events to load at once while selecting.",
+    ).tag(config=True)
+
+    output_path = traits.Path(
+        default_value="./IRF.fits.gz",
+        allow_none=False,
+        directory_ok=False,
+        help="Output file",
+    ).tag(config=True)
+
+    overwrite = Bool(
+        False,
+        help="Overwrite the output file if it exists",
+    ).tag(config=True)
+
+    obs_time = Float(default_value=50.0, help="Observation time").tag(config=True)
+    obs_time_unit = Unicode(
+        default_value="hour",
+        help="Unit used to specify observation time as an astropy unit string.",
+    ).tag(config=True)
+
+    alpha = Float(
+        default_value=0.2, help="Ratio between size of on and off regions"
+    ).tag(config=True)
+
+    classes = [
+        GridOptimizer,
+        SourceOffsetBinning,
+        OutputEnergyBinning,
+        EventPreProcessor,
+    ]
+
+    def setup(self):
+        self.opt_result = OptimisationResult()
+        self.theta = ThetaCutsCalculator(parent=self)
+        self.e_bins = OutputEnergyBinning(parent=self)
+        self.bins = SourceOffsetBinning(parent=self)
+        self.epp = EventPreProcessor(parent=self)
+
+        self.opt_result.read(self.cuts_file)
+        self.epp.quality_criteria = self.opt_result.precuts
+        self.reco_energy_bins = self.e_bins.reco_energy_bins()
+        self.true_energy_bins = self.e_bins.true_energy_bins()
+        self.energy_migration_bins = self.e_bins.energy_migration_bins()
+
+        self.source_offset_bins = self.bins.source_offset_bins()
+        self.fov_offset_bins = self.opt_result.offset_lim
+
+        self.particles = [
```

Comment:
Protons and electrons should probably be optional. You only need them to compute a background model, which we maybe should split into a separate tool (`ctapipe-compute-background-from-simulations`) or similar, since we won't use simulations for computing the background component later.